### ISSUE TITLE
SPV: Decorate the sampling instruction as RelaxedPrecision when when sampling a RelaxedPrecision sampler/image.

### ISSUE THREAD

**Issue by johnkslang** [KhronosGroup/glslang#1061]
Khronos decided (internal SPIR-V bug 181), regarding decorating the image type versus the sampling-instruction result, that:

>If either is decorated, then they both should be decorated and the decorations must match.  If an existing module has only one decorated then the result can use either precision and the driver can implement the relaxed precision behavior.

**Comment by johnkslang** [comment]
I quoted the wrong resolution to this issue.  The resolution was:

> Added to the list of things to decorate
> * The `Result <id>` of an instruction that reads or filters from an image. E.g. OpImageSampleExplicitLod.

> Add the paragraph:
> For image-sampling operations, decorations can appear on both the sampling instruction and the image variable being sampled. If either is decorated, they both should be decorated, and when both are decorated their decorations must match. If only one is decorated, the sampling instruction can behave either as if both were decorated or neither were decorated.

**Comment by johnkslang** [comment]
And, from KHR_vulkan_glsl:

> The precision qualification of the result of a built-in function is     determined in one of the following ways:

> - For the texture sampling, image load, and image store functions,          the precision of the return type matches the precision of the

```
            sampler type:
             uniform lowp sampler2D sampler;
             highp vec2 coord;
             ...
             lowp vec4 col = texture (sampler, coord); // texture() returns lowp
```

**Comment by johnkslang** [comment]
This was fixed by f0e35bf0ef68692412ff3d3ea0800957c3228e55 and fe4e572c532e8d894270040f35d2502057456361.
